feat: Rework Configurtion documentation for new Common Config (#978)
edgexfoundry/edgex-docs: docs_src/microservices/configuration/CommonConfiguration.md

# Common Configuration

The tables in each of the tabs below document configuration properties that are common to all services in the EdgeX Foundry platform. 
Service specific properties can be found on the respective documentation page for each service.

!!! edgey - "EdgeX 3.0"

>     New in Edgex 3.0 the common configuration is now in a single location, both source file and in the common section in the Configuration Provider. In prior releases the common configuration was duplicated in each service's configuration and section in the Configuration Provider.

Reviewer: you mention single location then indicate multiple. How about rewording it to this?
```suggestion
    New in Edgex 3.0 the common configuration is now in a single location. The source file is loaded into the common section in the Configuration Provider. In prior releases the common configuration was duplicated in each service's configuration and in each service's section in the Configuration Provider.
```

    ||Entries in the MessageBus section of the configuration allow for connecting to the internal MessageBus and define a common base topic prefix|
    |Protocol | redis| Indicates the connectivity protocol to use when connecting to the bus.|
    |Host | localhost | Indicates the host of the messaging broker, if applicable.|
    |Port | 6379| Indicates the port to use when publishing a message.|
    |Type | redis| Indicates the type of messaging library to use. Currently this is Redis by default. Refer to the [go-mod-messaging](https://github.com/edgexfoundry/go-mod-messaging) module for more information. |
    |AuthMode | usernamepassword| Auth Mode to connect to EdgeX MessageBus.|
    |SecretName | redisdb | Name of the secret in the Secret Store to find the MessageBus credentials.|
    |BaseTopicPrefix | edgex| Indicates the base topic prefix which is prepended to all internal MessageBus topics. |
=== "MessageQueue.Optional"
    |Property|Default Value|Description|
    |---|---|---|
    ||Configuration and connection parameters for use with MQTT or NATS message bus - in place of Redis|
    |ClientId| ---|Client ID used to put messages on the bus (specific for each service)|

>     |Qos|'0'| Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)|

Reviewer: ```suggestion
    |Qos|'0'| Quality of Service values are 0 (At most once), 1 (At least once) or 2 (Exactly once)|
```

>     |KeepAlive |'10'| Period of time in seconds to keep the connection alive when there is no messages flowing (must be 2 or greater)|

Reviewer: ```suggestion
    |KeepAlive |'10'| Period of time in seconds to keep the connection alive when there are no messages flowing (must be 2 or greater)|
```
